### PR TITLE
Refine MVAD sample

### PR DESCRIPTION
**Pull request by DavidZeLiang**
Refine MVAD sample to avoid potential blocking issues

### CODE REVIEW

File: sdk/anomalydetector/azure-ai-anomalydetector/samples/sample_multivariate_detect.py [hunk 1]
```diff
@@ -43,7 +44,7 @@ def __init__(self, subscription_key, anomaly_detector_endpoint, data_source=None
 
         self.data_source = data_source
 
-    def train(self, start_time, end_time, max_tryout=500):
+    def train(self, start_time, end_time, timeout=1000):
```
Comment (moreOver0) [inline sdk/anomalydetector/azure-ai-anomalydetector/samples/sample_multivariate_detect.py:47]: `timeout_in_second`

File: sdk/anomalydetector/azure-ai-anomalydetector/samples/sample_multivariate_detect.py [hunk 2]
```diff
@@ -62,21 +63,35 @@ def train(self, start_time, end_time, max_tryout=500):
...
+            time.sleep(1)
 
-        print("Done.", "\n--------------------")
-        print("{:d} available models after training.".format(len(new_model_list)))
+        if model_status == ModelStatus.FAILED:
+            print("Creating model failed.")
+            print("Errors:")
+            if model_info.errors:
+                for res in model_info.errors:
+                    print("Error code: {}. Message: {}".format(res.code, res.message))
```
Comment (moreOver0) [inline sdk/anomalydetector/azure-ai-anomalydetector/samples/sample_multivariate_detect.py:78]: `res` -> `error`

File: sdk/anomalydetector/azure-ai-anomalydetector/samples/sample_multivariate_detect.py [hunk 3]
```diff
@@ -62,21 +63,35 @@ def train(self, start_time, end_time, max_tryout=500):
...
+            else:
+                print("None")
+            return None
 
-        # Return the latest model id
-        return trained_model_id
+        if model_status == ModelStatus.READY:
+            print("Done.", "\n--------------------")
```
Comment (moreOver0) [inline sdk/anomalydetector/azure-ai-anomalydetector/samples/sample_multivariate_detect.py:84]: `print("Done.\n--------------------")`

Comment (moreOver0) [reply sdk/anomalydetector/azure-ai-anomalydetector/samples/sample_multivariate_detect.py:84]: movev the `new_model_list = list(self.ad_client.list_multivariate_model(skip=0, top=10000))` here

File: sdk/anomalydetector/azure-ai-anomalydetector/samples/sample_multivariate_detect.py [hunk 4]
```diff
@@ -62,21 +63,35 @@ def train(self, start_time, end_time, max_tryout=500):
...
 
-    def detect(self, model_id, start_time, end_time, max_tryout=500):
+            # Return the latest model id
+            return trained_model_id
+
+        # If model status is "CREATED" or "RUNNING"
+        print("Model is not ready yet. Model status: {}".format(model_status))
+        return None
+
+    def detect(self, model_id, start_time, end_time, timeout=500):
```
Comment (moreOver0) [inline sdk/anomalydetector/azure-ai-anomalydetector/samples/sample_multivariate_detect.py:94]: `timeout` -> `timeout_in_second`

**Comment by moreOver0** [comment]
@lmazuel @mayurid Could you help review this PR? We are trying to refine the SDK smple. The old version sample misleads our customers in some case.

**Review by lmazuel — APPROVED** [review]

**Pull request merged**